compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -300,6 +300,8 @@ trait ConstraintHandling[AbstractContext] {
    *      (i.e. `inst.widenSingletons <:< bound` succeeds with satisfiable constraint)
    *   2. If `inst` is a union type, approximate the union type from above by an intersection
    *      of all common base types, provided the result is a subtype of `bound`.
+   *   3. (currently not enabled) If `inst` is an intersection with some protected base types, drop
```

Comment:
I wouldn't use the word protected, unless the plan is to actually use the protected modifier with them. Maybe restricted ?